crates/kubelet/src/module_store.rs

```diff
@@ -12,6 +12,40 @@ use std::convert::TryFrom;
 use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
+/// Specifies how the store should check for module updates
+#[derive(PartialEq, Debug)]
+pub enum ModulePullPolicy {
+    /// Always return the module as it currently appears in the
+    /// upstream registry
+    Always,
+    /// Return the module as it is currently cached in the local store if
+    /// present; fetch it from the upstream registry only if it it not
+    /// present in the local store
+    IfNotPresent,
+    /// Never fetch the module from the upstream registry; if it is not
+    /// available locally then return an error
+    Never,
+}
+
+impl ModulePullPolicy {
+    /// Parses a module pull policy from a Kubernetes ImagePullPolicy string
+    pub fn parse(name: Option<String>) -> anyhow::Result<Option<ModulePullPolicy>> {
```

Comment:
I believe `IfNotPresent` is the default. Should we implement `Default` for `ModulePullPolicy` and then have this function just return `Result<ModulePullPolicy>`?

Reply:
If only it were that simple!  The default is `IfNotPresent` _if_ the reference contains a tag _and_ the tag is not `:latest`; otherwise it is `Always`.  (https://kubernetes.io/docs/concepts/configuration/overview/#container-images)  Hence the shenanigans to compute `effective_pull_policy` in `FileModuleStore::get`.